src/components/ec-phone-number-input/ec-phone-number-input.vue

```diff
@@ -149,197 +149,186 @@
-    },
+  bottomNote: {
+    type: String,
   },
-  methods: {
-    getPopoverStyle() {
-      if (this.$refs.popoverWidthReference) {
-        return {
-          width: `${this.$refs.popoverWidthReference.offsetWidth}px`,
-        };
-      }
-
-      return null;
-    },
-    onFocusCountry() {
-      this.countriesHasFocus = true;
-      this.$emit('focus');
-    },
-    onCountryChange(evt) {
-      this.countriesHasFocus = true;
-      this.$emit('change', evt);
-      this.$emit('country-change', evt);
-    },
-    onPhoneNumberChange(evt) {
-      this.$emit('change', evt);
-      this.$emit('phone-number-change', evt);
-    },
-    getCountryFlagPath(countryCode) {
-      if (!countryCode) {
-        return null;
-      }
-      try {
-        // eslint-disable-next-line global-require, import/no-dynamic-require
-        return require(`svg-country-flags/png100px/${countryCode.toLowerCase()}.png`);
-      } catch (err) {
-        return null;
-      }
+  isWarning: {
+    type: Boolean,
+    default: false,
+  },
+  isMasked: {
+    type: Boolean,
+    default: false,
+  },
+  warningTooltipMessage: {
+    type: String,
+  },
+  errorMessage: {
+    type: String,
+  },
+  errorTooltipMessage: {
+    type: String,
+  },
+  countries: {
+    type: Array,
+  },
+  areCountriesLoading: {
+    type: Boolean,
+    default: false,
+  },
+  isDisabled: {
+    type: Boolean,
+    default: false,
+  },
+  disabledTooltipMessage: {
+    type: String,
+  },
+  isSensitive: {
+    type: Boolean,
+    default: false,
+  },
+  countryPlaceholder: {
+    type: String,
+  },
+  phoneNumberPlaceholder: {
+    type: String,
+  },
+  isSearchEnabled: {
+    type: Boolean,
+    default: true,
+  },
+  searchCountryPlaceholder: {
+    type: String,
+    default: 'Search...',
+  },
+  noCountriesText: {
+    type: String,
+    default: 'No results found',
+  },
+  level: {
+    type: String,
+    validator(value) {
+      return ['notification', 'modal', 'tooltip', 'level-1', 'level-2', 'level-3'].includes(value);
     },
   },
-};
+});
+
+const emit = defineEmits([
+  'update:modelValue',
+  'change',
+  'focus',
+  'open',
+  'after-open',
+  'country-change',
+  'phone-number-change',
+]);
+
+const uid = getUid();
+const countriesHasFocus = ref(false);
+const popoverOptions = ref({
+  autoSize: 'min',
+});
+const popoverWidthReference = ref(null);
+
+const id = computed(() => `ec-phone-number-input-field-${uid}`);
+const errorId = computed(() => (isInvalid.value ? `ec-phone-number-input-${uid}` : null));
+const countriesItems = computed(() => props.countries.map(country => ({
+  areaCode: country.areaCode,
+  iconPath: getCountryFlagPath(country.countryCode),
+  id: country.countryCode,
+  name: country.text,
+  text: `${country.text} ${country.areaCode}`, // the text is only displayed in the tooltip, it's just to make items searchable by area code and name of the country
+  value: country,
+})));
+const countriesModel = computed({
```

Comment:
I would go even further and group stuff together. You can add comments to mark a sections too:

```
// ids
const uid
const id
const errorId

// items
const countriesItems
const selectedCountryAreaCode
const selectedCountryImage 
const selectedCountryName
function getCountryFlagPath(countryCode) {

// focus
const countriesHasFocus
function onFocusCountry() {

// models
const countriesModel = computed({
const phoneNumberModel = computed({
function onCountryChange(evt) {
function onPhoneNumberChange(evt) {

// popover
const popoverOptions = ref({
function getPopoverStyle() {
```

I think this is the benefit of having one setup function instead of the logic of the component scattered across the whole component like in the options API.

BTW, this component can be refactored in the future to support multiple v-models, that would be a breaking change which we don't want to add yet.

Reply:
ordered the script structure